### ISSUE TITLE
Error Downloading MicroG, Youtube Vanced, and Youtube Music on Android X86

### ISSUE THREAD

**Issue by Scottify20** [YTVanced/VancedManager#342]
- It's a laptop that originally runs Windows 10
- Operating System: Dual boot of Android x86_64 9.0 r2 with Windows 10
- Android Version: 9.0 x86_64 r2
- Vanced Manager Version: 2.1.0

Whenever i try to install MicroG it always shows "Error Downloading microg.apk"

I tried rebooting the device but it still shows the error

**Comment by X1nto** [comment]
Try clearing manager data

**Comment by Scottify20** [comment]
It still does not work

**Comment by X1nto** [comment]
If you have any firewalls, disable them

**Comment by Kyrex** [comment]
- Samsung Galaxy S10+ and Samsung Galaxy Tab S4
- Operating System: Android
- Android Version: 10
- Vanced Manager Version: 2.1.0
- Non-root option

I have the same issue since yesterday on two devices.

All of the apps seem like they lost internet connection (Manager/VancedYT/Music) so none of them work as of right now.
Manager shows all as Unavailable and cannot download or reinstall any of the apps.

Tried rebooting, clearing cache, different ISPs (Home and Work wifi), 4G LTE, Private VPN to three different countries. Nothing is working on both devices.

**Comment by X1nto** [comment]
> - Samsung Galaxy S10+ and Samsung Galaxy Tab S4
> - Operating System: Android
> - Android Version: 10
> - Vanced Manager Version: 2.1.0
> - Non-root option
> 
> I have the same issue since yesterday on two devices.
> 
> All of the apps seem like they lost internet connection (Manager/VancedYT/Music) so none of them work as of right now.
> Manager shows all as Unavailable and cannot download or reinstall any of the apps.
> 
> Tried rebooting, clearing cache, different ISPs (Home and Work wifi), 4G LTE, Private VPN to three different countries. Nothing is working on both devices.

Clearing cache won't help, you should clear whole data. Check if that works

**Comment by wrenhal** [comment]
I am on a Motorola zForce 2 running Android 8.  I previously was running version 2.0.2 of manager nonroot.
I referenced a push that said if you were running below version 2.2.1 to manually update from the web. I went into the app and everything said "unavailable" for "latest".
So I updated. Everything is still that way and nothing installs.  Just says "error downloading xxx.apk". Version of manager now shows as 2.2.1(rootedfirebase) even though I selected nonroot in the setup.
Did a force stop, cleared data, uninstalled, reinstalled. Same thing.  I have Blokada installed and I even set the app to bypass. My vanced app does still work also. 
I did.

**Comment by khaoskosmos** [comment]
Same problem on OnePlus 7 pro. Android 10.

**Comment by stefanseeger** [comment]
I had the same problem. It seems to be a problem with the file hosting. After changing nothing it just works again now.

**Comment by DarkendShadows** [comment]
Yep, same problem also on a OnePlus 7 Pro, mine is on the Android 11 Open Beta.
All options give a "Unavailable" or "Error Downloading (put apk name here).apk"

Vanced Manager Version: 2.2.1

**Comment by khaoskosmos** [comment]
There's solutions on the discord. Has nothing to do with clearing data etc. You have to enable dev mode in vanced manager and paste another url.

**Comment by X1nto** [comment]
> I am on a Motorola zForce 2 running Android 8.  I previously was running version 2.0.2 of manager nonroot.
> I referenced a push that said if you were running below version 2.2.1 to manually update from the web. I went into the app and everything said "unavailable" for "latest".
> So I updated. Everything is still that way and nothing installs.  Just says "error downloading xxx.apk". Version of manager now shows as 2.2.1(rootedfirebase) even though I selected nonroot in the setup.
> Did a force stop, cleared data, uninstalled, reinstalled. Same thing.  I have Blokada installed and I even set the app to bypass and turned it off. My vanced app does still work also. 
> I did.

Rooted firebase is the codename, it doesn't indicate if app is running in root mode or not 😑

**Comment by VALRINA-MAIN** [comment]
The website is down almost 24/7 that could be a problem...

**Comment by KevinX8** [comment]
yeah 24/7 definitely https://status.vancedapp.com/

**Comment by VALRINA-MAIN** [comment]
> yeah 24/7 definitely https://status.vancedapp.com/

Well then I am just unlocky lol. Anytime I want to update the site is down lol.

**Comment by willlgp** [comment]
!

**Comment by GMHR** [comment]
> yeah 24/7 definitely https://status.vancedapp.com/

It says the site and the API is functioning, but I still have the same issues

**Comment by wrenhal** [comment]
I will say that after updating the manager app again and waiting a day,
everything updated fine.

**Comment by X1nto** [comment]
@Scottify20 could you check in 2.4.0?

**Comment by fishcu** [comment]
@X1nto I still get the error that the apk (in my case "dark.apk" is the first one) cannot be downloaded with manager 2.4.0 and android x86.

**Comment by X1nto** [comment]
I'm pretty sure it's not architecture related but rather related to your internet, probably DNS poisoning or something like that

**Comment by X1nto** [comment]
@fishcu @Scottify20 if you still have problems, please consider using manager's built in Log menu and sending logs here (click the save button and send the latest log from /data/Android/com.vanced.manager/logs

**Comment by Arsen8349** [comment]
I on Redmi note 5 and vanced manager keep saying when I try download microg that it dont have presmission but i gave them to it(reinstalled around 5 times and cleared data 3 times, didnt work)

**Comment by fendis0709** [comment]
I used Redmi with MIUI and I managed to fix the issue.
Here's what I have done to fixes this issue:
- Go to Setting, select Manage Apps, and Enable "Storage Access" for Vanced Manager App
- Enable your Developer Options by tapping several times in your Android Version Info
- Go to Settings, select Developer Options, and turn off "MIUI Optimization" (Bottom of menu).
- Try update / reinstall again in Vanced App Manager
- If it's still not working, try install VPN. I used Intra DNS Encrypt (Available on Play Store).
.
By the way, I don't think removing cache or data could fix this issue, since the issue belongs to Android System, not the Vanced App.
Hope it helps.

**Comment by X1nto** [comment]
Please check on 2.6.0

**Comment by missiah** [comment]
Vanced Manager: 2.6.0
Android: 7.1.2 (nonroot)

For YT, YT Music & microG i always get download error.
I cannot manually add storage permission because the Manager did not ask for it once. https://imgur.com/a/A7odfhl

Internet working 100% on everything else.

Help!

**Comment by ghost** [comment]
> Vanced Manager: 2.6.0
> Android: 7.1.2 (nonroot)
> 
> For YT, YT Music & microG i always get download error.
> I cannot manually add storage permission because the Manager did not ask for it once. https://imgur.com/a/A7odfhl
> 
> Internet working 100% on everything else.
> 
> Help!

Manager doesn't need storage permission, it does everything in local app storage not the shared storage.
Go to the [Official Discord](https://discord.gg/vanced) for troubleshooting.

**Comment by X1nto** [comment]
If you are using google dns that might be a problem because when I myself used google dns it blocked downloading. Make sure you're not using any dns or switch to cloudflare if you insist on using one. Switching to cloudflare will also resolve problem for those who aren't using dns at all but are still getting errors.

**Comment by missiah** [comment]
I am using google dns, will switch to cloudflare and will report back soon.

**Comment by missiah** [comment]
Switching the DNS to Cloudflare did not help :(

**Comment by clawsl** [comment]
For anyone facing the download errors - I had a similar issue today, where it seemed VancedManager didn't have any file access permissions on my older rooted android 8 phone. What helped was the following for the non root app (the root app can't be installed this way, but non root is totally fine for me):

1. Install Micro G manually from https://github.com/YTVanced/VancedMicroG/releases (the apk is listed under assets and you need to enable "unknown sources" in settings)
2. Installl SAI  Split APK Installer from the Playstore - this app will be needed for combining the apks later
3. Download the latest 3 apk files from code mirror 
   - Architecture apk - https://mirror.codebucket.de/vanced/api/beta/apks/v16.16.38/nonroot/Arch/ contains the architecture apks (e.g. split_config.armeabi_v7a.apk is for arm32 and split_config.arm64_v8a.apk for arm64)
   -  Language `https://mirror.codebucket.de/vanced/api/beta/apks/v16.16.38/nonroot/Language/split_config.en.apk`
   - Theme apk `https://mirror.codebucket.de/vanced/api/beta/apks/v16.16.38/nonroot/Theme/dark.apk` (or black.apk if you prefer that)
4. Open SAI, click `Install APKs` and select the 3 apks you downloaded in 3. for installation 
5. Don't forget to contribute a bit to the maintenance of the [code mirror server](https://paypal.me/divadsn/2EUR)  (this small contribution goes to David, who hosts the files and not to me.) 
6. Enjoy your new Vanced version ;)
